draft-irtf-cfrg-vdaf.md

```diff
@@ -3593,6 +3593,101 @@ measurement is sharded. This is provided to the FLP by Prio3.
 | `Field`          | `Field128` ({{fields}})                         |
 {: title="Parameters of validity circuit Histogram."}
 
+### Prio3MultihotCountVec
+
+For this instance of Prio3, each measurement is a vector of ones and zeros,
+where the number of ones is bounded. This provides a functionality similar to
+Prio3Histogram except that more than one entry may be set. This allows
+Prio3MultihotCountVec to be composed with a randomized respjonse mechanism,
+like {{EPK14}}, for providing differential privacy. (For example, each Client
+would set each entry to one with some small probability.)
+
+Prio3MultihotCountVec uses XofTurboShake128 ({{xof-turboshake128}}) as its XOF.
+Its validity circuit, denoted `MultihotCountVec`, uses `Field128` ({{fields}})
+as its default finite field. It has three parameters: `length`, the number of
+of entries in the count vector; `max_weight`, the maximum weight of the vector
+(i.e., the number of non-zero entries); and `chunk_length`, which is used by by
+a circuit optimization described below. It is RECOMMENDED to set `chunk_length`
+to an integer near the square root of `length` (see
+{{parallel-sum-chunk-length}}).
+
+Encoding, truncation, and decoding are defined as follows:
+
+~~~
+def encode(self, measurement):
+    if len(measurement) != self.length:
+        raise ValueError('invalid Client measurement length')
+
+    # The first part is the vector of counters.
+    encoded = list(map(self.Field, measurement))
+
+    # The second part is the bit-encoded, offset weight.
+    offset_weight = self.offset + sum(encoded, self.Field(0))
+    encoded += self.Field.encode_into_bit_vector(
+        offset_weight.as_unsigned(), self.bits_for_weight)
+    return encoded
+
+def truncate(self, meas):
+    return meas[:self.length]
+
+def decode(self, output, _num_measurements):
+    return [bucket_count.as_unsigned() for bucket_count in output]
+~~~
+
+The `MultihotCountVec` circuit checks the encoded measurement has two
+properties. First, each entry is either a zero or a one. Second, the weight of
+the vector, when offset by `max_weight` is equal to the offset weight reported
```

Comment:
Related comment about `offset`: I don't think the circuit is offsetting by `max_weight`? The `offset` is the difference between `max_weight` and `next_power_of_2(max_weight) - 1`.